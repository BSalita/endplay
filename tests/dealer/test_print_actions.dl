produce 20

action 
  printall, 
  print(north, south), 
  printew, 
  printpbn, 
  printcompact hcp(north), 
  printoneline spades(west), 
  printes "Number of hearts: ", hearts(north), \n