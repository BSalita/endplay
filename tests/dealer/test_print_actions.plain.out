
              T74
              AKJ
              J7
              KJ743
QJ92                        K83
T3                          Q9654
KQ542                       96
62                          T85
              A65
              872
              AT83
              AQ9

              43
              AQ84
              KQ3
              JT63
6                           KT7
KT6                         97
J8765                       AT94
KQ72                        A954
              AQJ9852
              J532
              2
              8

              AT
              Q6
              T9752
              Q842
952                         J7
T752                        A984
AKJ63                       Q84
6                           KT73
              KQ8643
              KJ3
              ---
              AJ95

              A74
              KT4
              86
              Q7653
965                         KJ32
AQ763                       852
QT4                         A973
AT                          94
              QT8
              J9
              KJ52
              KJ82

              Q93
              QT3
              62
              Q7632
K875                        JT62
9874                        A65
754                         KQ3
K9                          JT8
              A4
              KJ2
              AJT98
              A54

              AJT7543
              A8
              A8
              92
62                          Q9
93                          KJT
QT65                        J97432
QJ874                       K6
              K8
              Q76542
              K
              AT53

              532
              AK87
              AJ8
              Q65
QJ86                        K97
J                           T92
Q642                        K97
AT42                        J873
              AT4
              Q6543
              T53
              K9

              Q86
              AK87
              A
              AKT32
T7543                       J9
64                          QJ953
K2                          T73
Q854                        J96
              AK2
              T2
              QJ98654
              7

              73
              A8
              A98
              KT8432
QT42                        K986
Q7                          T532
KJT72                       653
A6                          J9
              AJ5
              KJ964
              Q4
              Q75

              J6532
              JT
              Q654
              83
K                           Q987
K654                        A92
AK3                         JT2
QJ765                       K94
              AT4
              Q873
              987
              AT2

              AJ7643
              95
              A9
              Q43
T85                         K2
K                           876
J843                        Q52
AKT62                       J9875
              Q9
              AQJT432
              KT76
              ---

              84
              J873
              87653
              J8
AT652                       Q93
A52                         KT94
K92                         T
T6                          Q7543
              KJ7
              Q6
              AQJ4
              AK92

              KT64
              K32
              J76
              J98
J2                          AQ98
JT                          Q865
9852                        43
AKQT6                       532
              753
              A974
              AKQT
              74

              J852
              A2
              JT98
              A54
K97                         AT3
KJ                          QT98543
53                          72
KJT982                      Q
              Q64
              76
              AKQ64
              763

              AT752
              T2
              AK8
              542
93                          KQ
K3                          A984
QJ9543                      7
J83                         KQT976
              J864
              QJ765
              T62
              A

              A97
              JT9543
              A6
              75
KJT543                      Q6
Q                           K762
52                          KQ43
KJ84                        AT2
              82
              A8
              JT987
              Q963

              A84
              Q873
              T
              AKQ52
KQJT                        32
J96                         KT42
65                          KJ987
JT94                        76
              9765
              A5
              AQ432
              83

              AT9843
              ---
              QJ32
              973
K                           QJ7
K82                         Q9763
KT765                       8
AT64                        KQJ8
              652
              AJT54
              A94
              52

              KJ9
              875
              K63
              J765
T7                          AQ3
QT964                       A2
T875                        AQJ942
QT                          84
              86542
              KJ3
              ---
              AK932

              965
              A943
              K
              98732
Q432                        AJ8
QT75                        J862
Q75                         A984
A5                          K6
              KT7
              K
              JT632
              QJT4

              T74
              AKJ
              J7
              KJ743
                            
                            
                            
                            
              A65
              872
              AT83
              AQ9

              43
              AQ84
              KQ3
              JT63
                            
                            
                            
                            
              AQJ9852
              J532
              2
              8

              AT
              Q6
              T9752
              Q842
                            
                            
                            
                            
              KQ8643
              KJ3
              ---
              AJ95

              A74
              KT4
              86
              Q7653
                            
                            
                            
                            
              QT8
              J9
              KJ52
              KJ82

              Q93
              QT3
              62
              Q7632
                            
                            
                            
                            
              A4
              KJ2
              AJT98
              A54

              AJT7543
              A8
              A8
              92
                            
                            
                            
                            
              K8
              Q76542
              K
              AT53

              532
              AK87
              AJ8
              Q65
                            
                            
                            
                            
              AT4
              Q6543
              T53
              K9

              Q86
              AK87
              A
              AKT32
                            
                            
                            
                            
              AK2
              T2
              QJ98654
              7

              73
              A8
              A98
              KT8432
                            
                            
                            
                            
              AJ5
              KJ964
              Q4
              Q75

              J6532
              JT
              Q654
              83
                            
                            
                            
                            
              AT4
              Q873
              987
              AT2

              AJ7643
              95
              A9
              Q43
                            
                            
                            
                            
              Q9
              AQJT432
              KT76
              ---

              84
              J873
              87653
              J8
                            
                            
                            
                            
              KJ7
              Q6
              AQJ4
              AK92

              KT64
              K32
              J76
              J98
                            
                            
                            
                            
              753
              A974
              AKQT
              74

              J852
              A2
              JT98
              A54
                            
                            
                            
                            
              Q64
              76
              AKQ64
              763

              AT752
              T2
              AK8
              542
                            
                            
                            
                            
              J864
              QJ765
              T62
              A

              A97
              JT9543
              A6
              75
                            
                            
                            
                            
              82
              A8
              JT987
              Q963

              A84
              Q873
              T
              AKQ52
                            
                            
                            
                            
              9765
              A5
              AQ432
              83

              AT9843
              ---
              QJ32
              973
                            
                            
                            
                            
              652
              AJT54
              A94
              52

              KJ9
              875
              K63
              J765
                            
                            
                            
                            
              86542
              KJ3
              ---
              AK932

              965
              A943
              K
              98732
                            
                            
                            
                            
              KT7
              K
              JT632
              QJT4

              
              
              
              
QJ92                        K83
T3                          Q9654
KQ542                       96
62                          T85
              
              
              
              

              
              
              
              
6                           KT7
KT6                         97
J8765                       AT94
KQ72                        A954
              
              
              
              

              
              
              
              
952                         J7
T752                        A984
AKJ63                       Q84
6                           KT73
              
              
              
              

              
              
              
              
965                         KJ32
AQ763                       852
QT4                         A973
AT                          94
              
              
              
              

              
              
              
              
K875                        JT62
9874                        A65
754                         KQ3
K9                          JT8
              
              
              
              

              
              
              
              
62                          Q9
93                          KJT
QT65                        J97432
QJ874                       K6
              
              
              
              

              
              
              
              
QJ86                        K97
J                           T92
Q642                        K97
AT42                        J873
              
              
              
              

              
              
              
              
T7543                       J9
64                          QJ953
K2                          T73
Q854                        J96
              
              
              
              

              
              
              
              
QT42                        K986
Q7                          T532
KJT72                       653
A6                          J9
              
              
              
              

              
              
              
              
K                           Q987
K654                        A92
AK3                         JT2
QJ765                       K94
              
              
              
              

              
              
              
              
T85                         K2
K                           876
J843                        Q52
AKT62                       J9875
              
              
              
              

              
              
              
              
AT652                       Q93
A52                         KT94
K92                         T
T6                          Q7543
              
              
              
              

              
              
              
              
J2                          AQ98
JT                          Q865
9852                        43
AKQT6                       532
              
              
              
              

              
              
              
              
K97                         AT3
KJ                          QT98543
53                          72
KJT982                      Q
              
              
              
              

              
              
              
              
93                          KQ
K3                          A984
QJ9543                      7
J83                         KQT976
              
              
              
              

              
              
              
              
KJT543                      Q6
Q                           K762
52                          KQ43
KJ84                        AT2
              
              
              
              

              
              
              
              
KQJT                        32
J96                         KT42
65                          KJ987
JT94                        76
              
              
              
              

              
              
              
              
K                           QJ7
K82                         Q9763
KT765                       8
AT64                        KQJ8
              
              
              
              

              
              
              
              
T7                          AQ3
QT964                       A2
T875                        AQJ942
QT                          84
              
              
              
              

              
              
              
              
Q432                        AJ8
QT75                        J862
Q75                         A984
A5                          K6
              
              
              
              

N:T74.AKJ.J7.KJ743 K83.Q9654.96.T85 A65.872.AT83.AQ9 QJ92.T3.KQ542.62
N:43.AQ84.KQ3.JT63 KT7.97.AT94.A954 AQJ9852.J532.2.8 6.KT6.J8765.KQ72
N:AT.Q6.T9752.Q842 J7.A984.Q84.KT73 KQ8643.KJ3..AJ95 952.T752.AKJ63.6
N:A74.KT4.86.Q7653 KJ32.852.A973.94 QT8.J9.KJ52.KJ82 965.AQ763.QT4.AT
N:Q93.QT3.62.Q7632 JT62.A65.KQ3.JT8 A4.KJ2.AJT98.A54 K875.9874.754.K9
N:AJT7543.A8.A8.92 Q9.KJT.J97432.K6 K8.Q76542.K.AT53 62.93.QT65.QJ874
N:532.AK87.AJ8.Q65 K97.T92.K97.J873 AT4.Q6543.T53.K9 QJ86.J.Q642.AT42
N:Q86.AK87.A.AKT32 J9.QJ953.T73.J96 AK2.T2.QJ98654.7 T7543.64.K2.Q854
N:73.A8.A98.KT8432 K986.T532.653.J9 AJ5.KJ964.Q4.Q75 QT42.Q7.KJT72.A6
N:J6532.JT.Q654.83 Q987.A92.JT2.K94 AT4.Q873.987.AT2 K.K654.AK3.QJ765
N:AJ7643.95.A9.Q43 K2.876.Q52.J9875 Q9.AQJT432.KT76. T85.K.J843.AKT62
N:84.J873.87653.J8 Q93.KT94.T.Q7543 KJ7.Q6.AQJ4.AK92 AT652.A52.K92.T6
N:KT64.K32.J76.J98 AQ98.Q865.43.532 753.A974.AKQT.74 J2.JT.9852.AKQT6
N:J852.A2.JT98.A54 AT3.QT98543.72.Q Q64.76.AKQ64.763 K97.KJ.53.KJT982
N:AT752.T2.AK8.542 KQ.A984.7.KQT976 J864.QJ765.T62.A 93.K3.QJ9543.J83
N:A97.JT9543.A6.75 Q6.K762.KQ43.AT2 82.A8.JT987.Q963 KJT543.Q.52.KJ84
N:A84.Q873.T.AKQ52 32.KT42.KJ987.76 9765.A5.AQ432.83 KQJT.J96.65.JT94
N:AT9843..QJ32.973 QJ7.Q9763.8.KQJ8 652.AJT54.A94.52 K.K82.KT765.AT64
N:KJ9.875.K63.J765 AQ3.A2.AQJ942.84 86542.KJ3..AK932 T7.QT964.T875.QT
N:965.A943.K.98732 AJ8.J862.A984.K6 KT7.K.JT632.QJT4 Q432.QT75.Q75.A5
north         east          south         west         

S T74         S K83         S A65         S QJ92        13           
H AKJ         H Q9654       H 872         H T3                       
D J7          D 96          D AT83        D KQ542                    
C KJ743       C T85         C AQ9         C 62                       

S 43          S KT7         S AQJ9852     S 6           12           
H AQ84        H 97          H J532        H KT6                      
D KQ3         D AT94        D 2           D J8765                    
C JT63        C A954        C 8           C KQ72                     

S AT          S J7          S KQ8643      S 952         8            
H Q6          H A984        H KJ3         H T752                     
D T9752       D Q84         D             D AKJ63                    
C Q842        C KT73        C AJ95        C 6                        

S A74         S KJ32        S QT8         S 965         9            
H KT4         H 852         H J9          H AQ763                    
D 86          D A973        D KJ52        D QT4                      
C Q7653       C 94          C KJ82        C AT                       

S Q93         S JT62        S A4          S K875        6            
H QT3         H A65         H KJ2         H 9874                     
D 62          D KQ3         D AJT98       D 754                      
C Q7632       C JT8         C A54         C K9                       

S AJT7543     S Q9          S K8          S 62          13           
H A8          H KJT         H Q76542      H 93                       
D A8          D J97432      D K           D QT65                     
C 92          C K6          C AT53        C QJ874                    

S 532         S K97         S AT4         S QJ86        14           
H AK87        H T92         H Q6543       H J                        
D AJ8         D K97         D T53         D Q642                     
C Q65         C J873        C K9          C AT42                     

S Q86         S J9          S AK2         S T7543       20           
H AK87        H QJ953       H T2          H 64                       
D A           D T73         D QJ98654     D K2                       
C AKT32       C J96         C 7           C Q854                     

S 73          S K986        S AJ5         S QT42        11           
H A8          H T532        H KJ964       H Q7                       
D A98         D 653         D Q4          D KJT72                    
C KT8432      C J9          C Q75         C A6                       

S J6532       S Q987        S AT4         S K           4            
H JT          H A92         H Q873        H K654                     
D Q654        D JT2         D 987         D AK3                      
C 83          C K94         C AT2         C QJ765                    

S AJ7643      S K2          S Q9          S T85         11           
H 95          H 876         H AQJT432     H K                        
D A9          D Q52         D KT76        D J843                     
C Q43         C J9875       C             C AKT62                    

S 84          S Q93         S KJ7         S AT652       2            
H J873        H KT94        H Q6          H A52                      
D 87653       D T           D AQJ4        D K92                      
C J8          C Q7543       C AK92        C T6                       

S KT64        S AQ98        S 753         S J2          8            
H K32         H Q865        H A974        H JT                       
D J76         D 43          D AKQT        D 9852                     
C J98         C 532         C 74          C AKQT6                    

S J852        S AT3         S Q64         S K97         10           
H A2          H QT98543     H 76          H KJ                       
D JT98        D 72          D AKQ64       D 53                       
C A54         C Q           C 763         C KJT982                   

S AT752       S KQ          S J864        S 93          11           
H T2          H A984        H QJ765       H K3                       
D AK8         D 7           D T62         D QJ9543                   
C 542         C KQT976      C A           C J83                      

S A97         S Q6          S 82          S KJT543      9            
H JT9543      H K762        H A8          H Q                        
D A6          D KQ43        D JT987       D 52                       
C 75          C AT2         C Q963        C KJ84                     

S A84         S 32          S 9765        S KQJT        15           
H Q873        H KT42        H A5          H J96                      
D T           D KJ987       D AQ432       D 65                       
C AKQ52       C 76          C 83          C JT94                     

S AT9843      S QJ7         S 652         S K           7            
H             H Q9763       H AJT54       H K82                      
D QJ32        D 8           D A94         D KT765                    
C 973         C KQJ8        C 52          C AT64                     

S KJ9         S AQ3         S 86542       S T7          8            
H 875         H A2          H KJ3         H QT964                    
D K63         D AQJ942      D             D T875                     
C J765        C 84          C AK932       C QT                       

S 965         S AJ8         S KT7         S Q432        7            
H A943        H J862        H K           H QT75                     
D K           D A984        D JT632       D Q75                      
C 98732       C K6          C QJT4        C A5                       

T74.AKJ.J7.KJ743 K83.Q9654.96.T85 A65.872.AT83.AQ9 QJ92.T3.KQ542.62 4
43.AQ84.KQ3.JT63 KT7.97.AT94.A954 AQJ9852.J532.2.8 6.KT6.J8765.KQ72 1
AT.Q6.T9752.Q842 J7.A984.Q84.KT73 KQ8643.KJ3..AJ95 952.T752.AKJ63.6 3
A74.KT4.86.Q7653 KJ32.852.A973.94 QT8.J9.KJ52.KJ82 965.AQ763.QT4.AT 3
Q93.QT3.62.Q7632 JT62.A65.KQ3.JT8 A4.KJ2.AJT98.A54 K875.9874.754.K9 4
AJT7543.A8.A8.92 Q9.KJT.J97432.K6 K8.Q76542.K.AT53 62.93.QT65.QJ874 2
532.AK87.AJ8.Q65 K97.T92.K97.J873 AT4.Q6543.T53.K9 QJ86.J.Q642.AT42 4
Q86.AK87.A.AKT32 J9.QJ953.T73.J96 AK2.T2.QJ98654.7 T7543.64.K2.Q854 5
73.A8.A98.KT8432 K986.T532.653.J9 AJ5.KJ964.Q4.Q75 QT42.Q7.KJT72.A6 4
J6532.JT.Q654.83 Q987.A92.JT2.K94 AT4.Q873.987.AT2 K.K654.AK3.QJ765 1
AJ7643.95.A9.Q43 K2.876.Q52.J9875 Q9.AQJT432.KT76. T85.K.J843.AKT62 3
84.J873.87653.J8 Q93.KT94.T.Q7543 KJ7.Q6.AQJ4.AK92 AT652.A52.K92.T6 5
KT64.K32.J76.J98 AQ98.Q865.43.532 753.A974.AKQT.74 J2.JT.9852.AKQT6 2
J852.A2.JT98.A54 AT3.QT98543.72.Q Q64.76.AKQ64.763 K97.KJ.53.KJT982 3
AT752.T2.AK8.542 KQ.A984.7.KQT976 J864.QJ765.T62.A 93.K3.QJ9543.J83 2
A97.JT9543.A6.75 Q6.K762.KQ43.AT2 82.A8.JT987.Q963 KJT543.Q.52.KJ84 6
A84.Q873.T.AKQ52 32.KT42.KJ987.76 9765.A5.AQ432.83 KQJT.J96.65.JT94 4
AT9843..QJ32.973 QJ7.Q9763.8.KQJ8 652.AJT54.A94.52 K.K82.KT765.AT64 1
KJ9.875.K63.J765 AQ3.A2.AQJ942.84 86542.KJ3..AK932 T7.QT964.T875.QT 2
965.A943.K.98732 AJ8.J862.A984.K6 KT7.K.JT632.QJT4 Q432.QT75.Q75.A5 4
Number of hearts: 3

Number of hearts: 4

Number of hearts: 2

Number of hearts: 3

Number of hearts: 3

Number of hearts: 2

Number of hearts: 4

Number of hearts: 4

Number of hearts: 2

Number of hearts: 2

Number of hearts: 2

Number of hearts: 4

Number of hearts: 3

Number of hearts: 2

Number of hearts: 2

Number of hearts: 6

Number of hearts: 4

Number of hearts: 0

Number of hearts: 3

Number of hearts: 4


